internal/vpn.go

```diff
@@ -115,17 +115,30 @@ func updateConfig(app *App, client *http.Client, pubkey string) error {
 func initVpn(app *App, client *http.Client, crypto CryptoHandler) error {
 	sotaConfig := filepath.Dir(app.EncryptedConfig)
 	wgPriv := filepath.Join(sotaConfig, "wg-priv")
+	register := false
 	if _, err := os.Stat(wgPriv); os.IsNotExist(err) {
+		register = true
 		log.Println("Wireguard private key does not exist, generating.")
-		pub, err := generateKey(wgPriv)
+	} else {
+		secretFile := filepath.Join(app.SecretsDir, "wireguard-client")
+		if _, err := os.Stat(secretFile); os.IsNotExist(err) {
+			register = true
+			log.Println("Wireguard key not registered on server, will re-register")
+		}
+	}
+	if register {
+		wgPrivTmp := wgPriv + ".tmp"
+		pub, err := generateKey(wgPrivTmp)
```

Comment:
If we the file is available but not submitted we will end up generating a new key, but I guess it is fine because we're replacing later with rename?

Reply:
correct. wasted cpu time, but not too significant.